Source/Immutable/Public/Immutable/ImmutablePluginSettings.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "Engine/DeveloperSettings.h"
+#include "ApplicationConfig.h"
+
+#include "ImmutablePluginSettings.generated.h"
+
+
+/**
+ * ImmutablePluginSettings is a configuration class for the Immutable plugin.
+ * This class contains settings that can be adjusted to control the behavior
+ * of the Immutable plugin within the Unreal Engine environment.
+ */
+UCLASS(config = Game, defaultconfig, meta = (DisplayName = "Immutable Plugin Settings"))
+class IMMUTABLE_API UImmutablePluginSettings : public UDeveloperSettings
+{
+	GENERATED_BODY()
+
+public:
+	/// The default application configuration class.
+	/// This property holds a reference to a subclass of UApplicationConfig,
+	/// which will be used as the default configuration for the application.
+	UPROPERTY(Config, EditAnywhere, BlueprintReadOnly, Category = "General")
+	TSubclassOf<UApplicationConfig> DefaultApplicationConfig;

```

Comment:
I think what you're trying to do here is:
Having persistent a `DefaultApplicationConfig` that will be serialized/saved. Hence the `Config`
But this won't work because of how you have it set up

You will need to change `TSubclassOf<UApplicationConfig> DefaultApplicationConfig;` to, `FApplicationConfig DefaultApplicationConfig;`
And `UApplicationConfig` from a `UCLASS` to a `USTRUCT` (and renaming `UApplicationConfig` to `FApplicationConfig`) otherwise what you're trying to do with `UPROPERTY(Config ...)` won't work

Unless, what you're trying to do is
Creating a Blueprint asset which `DefaultApplicationConfig` will reference
But I think this way creates hassle and more room for error

Reply:
Not really sure on the point that it won't work. Could you elaborate in details. I have tested it succefully. 

On the hassle point, we have to do it this way instead of for example having just everything in the developer settings because two main reasons:
1. github/repo protection will complain that devs store  passwords and valuable info
2. The app config won't be accessible from the text editors